synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -1498,3 +1506,96 @@ def _populate_txn(txn: LoggingTransaction) -> bool:
             )
 
         return batch_size
+
+    async def _populate_membership_event_stream_ordering(
+        self, progress: JsonDict, batch_size: int
+    ) -> int:
+        def _populate_membership_event_stream_ordering(
+            txn: LoggingTransaction,
+        ) -> bool:
+            if "max_stream_ordering" in progress:
+                max_stream_ordering = progress["max_stream_ordering"]
```

Comment:
Note that if somebody:
 1. upgrades their Synapse to a version that supports this change
 2. rolls back for a while
 3. upgrades again

... then we will end up missing a number of events.

The only real solution is what we did for `events_populate_state_key_rejections`: make sure that the code to populate the columns for new rows lands well before the code to back-populate existing rows, and do the `SCHEMA_VERSION`/`SCHEMA_COMPAT_VERSION` dance to stop people rolling back too far.

In theory we can document it as a thing that people need to watch out for, but people don't read documentation.

Reply:
Hm yes that’s a great point, people never read documentation :)! I will split this again and make this PR only populate the field and kick the background update for later inclusion with the schema dance.

Reply:
Reverted in https://github.com/matrix-org/synapse/pull/15128/commits/845c9b90ff669a2a7394cbeb93d4b0f338a78af1, will make a new PR for the bg jobs!